src/topic_training/src/Demo.cpp

```diff
@@ -0,0 +1,136 @@
+/* Include all headers of the ROS system we are using 
+   separately. */
+#include "ros/init.h"
+#include "ros/node_handle.h"
+#include "ros/rate.h"
+#include "ros/subscriber.h"
+#include "ros/publisher.h"
+
+/* Include the message type we will be publishing. */
+#include "std_msgs/String.h"
+
+/* Include Stringstream to be able to manipulate the 
+   data we are sending and receiving. (Optional: only 
+   if the message is of type String)*/
+#include <sstream>
+
+/* C++ does not like "magic numbers," so it is better 
+   if we make them variables. */
+const int QUEUE_SIZE = 1000;
+const int FREQUENCY = 10;
+
+/**
+ * This function will get called when a new message has 
+ * arrived on the topic.
+ * 
+ * @param msg - the message received by the Subscriber, 
+ *              sent by the Publisher
+ */
+void chatterCallback(const std_msgs::String::ConstPtr& msg)
+{
+   /* Print out the message to the terminal. */
+   ROS_INFO("I heard: [%s]", msg->data.c_str());
+}
+
+/**
+ * The main method that contains the creation of the 
+ * node, Publisher, Subcriber, and topic. It also 
+ * describes what we will do with the data/message.
+ */
+auto main(int argc, char **argv) -> int
```

Comment:
Avoid this for the demo file.
```suggestion
int main(int argc, char **argv)
```

Reply:
Fixed in d3a5f8a.